### PR TITLE
Thermomachine have full temperature range on both modes

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
See title, 
both cooler and heater will have both ranges on min and max temperature (on t1 they can both go from 73 K to 573 K)
The thermomachine have a new function, the automatic thermal regulation where it will try to keep the temperature to the set one by automatically switching between heater and cooler; it won't change between using enviroment heat

## Why It's Good For The Game
Quality of life for players
Thermomachines are now able to sustain a temperature on their own; this will help with the new rework so that the already placed machines will automatically reach the best temperature even if the pipenet's temperature changes. Also players are able to create automatic gas production setups where the temperature may change too much due to the energy released by the gas created

## Changelog
:cl:
qol: Thermomachine have full temperature range on both modes, plus new automatic thermal regulator function.
/:cl:

### CODE REVIEW

**Comment by Jhyrachy** [comment]
Why not removing the switch?  

It kinda had sense when it deleted heat, but now that it moves it around (or produce it with electricity) isn't just simpler to set a temperature and the machine will try to change it to desired one?

**Comment by Ghilker** [comment]
> Why not removing the switch?
> 
> It kinda had sense when it deleted heat, but now that it moves it around (or produce it with electricity) isn't just simpler to set a temperature and the machine will try to change it to desired one?

i may do that if this get merged

**Comment by Jhyrachy** [comment]
It would also helps declutter a bit, since now is pretty cramped with the new settings

**Review by stylemistake — APPROVED** [review]

File: code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm [hunk 1]
```diff
@@ -143,7 +134,10 @@
 	var/main_heat_capacity = main_port.heat_capacity()
 	var/thermal_heat_capacity = thermal_exchange_port.heat_capacity()
 	var/temperature_delta = main_port.temperature - target_temperature
-	temperature_delta = cooling ? max(temperature_delta, 0) : min(temperature_delta, 0) //no cheesy strats
+	if(auto_thermal_regulator)
+		cooling = temperature_delta > 0 ? TRUE : FALSE
```
Comment (Mothblocks) [inline code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm:138]:
```suggestion
		cooling = temperature_delta > 0
```

**Comment by LemonInTheDark** [comment]
Could you please elaborate on why this is good for the game?

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**